File: go/feature.go
// this file is @generated
package meteroid

import "time"

type Feature struct {
	// Unique key used to reference this feature in your code. Cannot be changed after creation.
	Code string `json:"code"`

	// RFC 3339 timestamp.
	CreatedAt time.Time `json:"created_at"`

	Description *string `json:"description,omitempty"`

	Entitlement *Entitlement `json:"entitlement,omitempty"`

	FeatureType FeatureType `json:"feature_type"`

	Id FeatureId `json:"id"`

	Name string `json:"name"`

	Product *EntitlementProductRef `json:"product,omitempty"`

	Status FeatureStatus `json:"status"`
}

File: python/meteroid/models/feature.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .entitlement import Entitlement
from .entitlement_product_ref import EntitlementProductRef
from .feature_id import FeatureId
from .feature_status import FeatureStatus
from .feature_type import FeatureType


@dataclasses.dataclass
class Feature(BaseModel):
    code: str
    """Unique key used to reference this feature in your code. Cannot be changed after creation."""

    created_at: datetime

    feature_type: FeatureType

    id: FeatureId

    name: str

    status: FeatureStatus

    description: t.Optional[str] = None

    entitlement: t.Optional[Entitlement] = None

    product: t.Optional[EntitlementProductRef] = None
